docs/source/howto/codes.rst

```diff
@@ -4,41 +4,265 @@
 How to run external codes
 *************************
 
+To run an external code with AiiDA, you will need to use an appropriate :ref:`calculation plugin <topics:plugins>`.
+Its job is to orchestrate the whole process that starts with transforming the input nodes to the input files that the code expects, then copy everything to the machine where the code is located (using a :ref:`transport plugin <how-to:codes:transport>`), run the actual code (using a :ref:`scheduler plugin <how-to:codes:scheduler>`), and finally get back the outputs and, optionally, parse them into queryable AiiDA nodes (using a :ref:`parser plugin <how-to:codes:parser>`).
 
-.. _how-to:codes:plugin:
+The following subsections will take you through the process of :ref:`creating the calculation plugin<how-to:codes:interfacing>` and its acompanying :ref:`parser<how-to:codes:parsing>`, and then using these to actually :ref:`run the code<how-to:codes:run>`.
+Other supporting sections are also included.
```

Comment:
After reviewing this once more, I think the "Design guidelines" from line 144 should go right after this paragraph.

Reply:
Ok, I can change it. The only caveat is that this are designing guidelines, and although most sections are about designing (calcjobs, parsers, adding caching, schedulers, transports), there are some that are not (setting up computer and running), and furthermore the whole section is called "How to **run** external codes", so it might be weird to read about **design** guidelines in the begining. Please confirm if this is ok.

Reply:
Yeah, let's just go with it for now.